test/nbrowser/VersionUpdateBanner.ts

```diff
@@ -0,0 +1,86 @@
+import {assert, driver} from 'mocha-webdriver';
+
+describe('VersionUpdateBanner', function() {
+  this.timeout(300000);
+  setupTestSuite();
+
+  let oldEnv: testUtils.EnvironmentSnapshot;
+  let session: gu.Session;
+  let fakeServer: FakeUpdateServer;
+
+  afterEach(() => gu.checkForErrors());
+
+  before(async function() {
+    oldEnv = new testUtils.EnvironmentSnapshot();
+    process.env.GRIST_ALLOW_AUTOMATIC_VERSION_CHECKING = 'true';
+    process.env.GRIST_TEST_IMMEDIATE_VERSION_CHECK = 'true';
+    process.env.GRIST_DEFAULT_EMAIL = gu.session().email;
+    fakeServer = await startFakeUpdateServer();
+    process.env.GRIST_TEST_VERSION_CHECK_URL = `${fakeServer.url()}/version`;
+
+    // The fake server sets its payload after it has served a version
+    // at least once.
+    assert.isNull(fakeServer.payload);
+    await server.restart(true);
+    // Let's check it really was called now.
+    assert.isNotNull(fakeServer.payload);
+  });
+
+  after(async function() {
+    await fakeServer.close();
+    oldEnv.restore();
+    await server.restart(true);
+  });
+
+  it('should not be shown to non-managers', async () => {
+    session = await gu.session().user('user2').personalSite.login();
+    await session.loadDocMenu('/');
+    await driver.executeScript('window.localStorage.clear();');
+
+    await driver.findWait('.test-top-panel', 100);
+    assert.equal((await driver.findAll('.test-version-update-banner-text')).length, 0);
+  });
+
+  it('should be shown to managers', async () => {
+    session = await gu.session().personalSite.login();
+    await session.loadDocMenu('/');
+    await driver.executeScript('window.localStorage.clear();');
+
+    await driver.findWait('.test-top-panel', 100);
+    assert.equal(await driver.find('.test-version-update-banner-text').isDisplayed(), true);
+
+    // Should be dismissable
+    await driver.find('.test-banner-close').click();
+    assert.equal((await driver.findAll('.test-version-update-banner-text')).length, 0);
+
+    // Let's make sure it's still not there after being dismissed once
+    await driver.navigate().refresh();
+    await session.loadDocMenu('/');
+    await driver.findWait('.test-top-panel', 100);
+    assert.equal((await driver.findAll('.test-version-update-banner-text')).length, 0);
+
+    // Update the version, the banner should come back
+    fakeServer.bumpVersion();
+    await server.restart(false);
+    session = await gu.session().personalSite.login();
+    await session.loadDocMenu('/');
+    assert.equal(await driver.find('.test-version-update-banner-text').isDisplayed(), true);
+  });
+
+  it('should not be shown to managers when there is no newer version', async () => {
+    fakeServer.latestVersion = installedVersion;
+    await server.restart(true);
+    session = await gu.session().personalSite.login();
+    await session.loadDocMenu('/');
+    await driver.executeScript('window.localStorage.clear();');
```

Comment:
Isn't it too late to clear the locale storage? I think it is better to do it before some browser load (or at least before refreshing the page)

Reply:
Okay, I've moved it to before the page load.

Reply:
Is there any commit I can check what you did after my review? Last commit is from May 7.